prov/tcp/src/tcpx_ep.c

```diff
@@ -136,8 +141,10 @@ static ssize_t tcpx_sendmsg(struct fid_ep *ep, const struct fi_msg *msg,
 	data_len = ofi_total_iov_len(msg->msg_iov, msg->iov_count);
 
 	if (flags & FI_INJECT) {
-		if (data_len > TCPX_MAX_INJECT_SZ)
-			return -FI_EINVAL;
+		if (data_len > TCPX_MAX_INJECT_SZ) {
+			ret = -FI_EINVAL;
+			goto err;
+		}
```

Comment:
This should be an assert() rather than a runtime check.  The app has the inject size at initialization and should ensure that it limits transfers appropriately.